eip-5564.md

```diff
@@ -8,174 +8,170 @@ status: Draft
 type: Standards Track
 category: ERC
 created: 2022-08-13
-requires: 165
 ---
 
 
 ## Abstract
-This specification defines a standardized way of creating stealth addresses. This EIP enables senders of transactions/transfers to non-interactively generate private stealth addresses for their recipients that only the recipients can unlock. 
+This specification defines a standardized way of creating stealth addresses. This EIP enables senders of transactions/transfers to non-interactively generate private stealth addresses for their recipients that only the recipients can unlock.
 
 ## Motivation
-The standardization of non-interactive stealth address generation may unlock significant privacy potential of Ethereum by allowing the recipient of a transfer to remain private when receiving an asset. A stealth address is generated by the sender from a shared secret between sender and recipient and can only be unlocked by the recipient. The reseason for that is, that only the recipient can compute the private key that eventually unlocks the funds at the stealth address.
+The standardization of non-interactive stealth address generation may unlock significant privacy potential of Ethereum by allowing the recipient of a transfer to remain private when receiving an asset. A stealth address is generated by the sender from a shared secret between sender and recipient and can only be unlocked by the recipient. The reason for that is, that only the recipient can compute the private key that eventually unlocks the funds at the stealth address.
 Observers have no possibility of linking the recipient's stealth address to the recipient's identity, leaving only the sender with that information.
 
 ## Specification
 The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
 
-Every Smart Contract Wallet compliant with this EIP MUST implement the [EIP-165](./eip-165.md) (`0x01ffc9a7`) interface.
+The follow contracts are part of this specification:
+- `IERC5564Registry` stores the stealth public keys for users. This MUST be a singleton contract, with one instance per chain.
+- `IERC5565Generator` contracts are used to compute stealth addresses for a user based on a given curve. There can be many of these per chain, and for a given curve there SHOULD be one implementation per chain.
+- `IERC5564Messenger` emits events to announce when something is sent to a stealth address. This MUST be a singleton contract, with one instance per chain.
+
+The interface for each is specified as follows:
```

Comment:
The Solidity below has comments containing requirements for each method. We may need to pull these out into real text for the actual EIP depending on what the EIP editors say, but for now I think this is a clear, readable to document function-specific requirements.